File: java-algorithms/src/main/java/com/omsprog/algorithms/lc/FizzBuzz.java
package com.omsprog.algorithms.lc;

import java.util.ArrayList;
import java.util.List;

public class FizzBuzz {
    public List<String> fizzBuzz(int n) {
        List<String> list = new ArrayList<>(n);

        for(int i = 1; i <= n; i++) {

            StringBuilder sb = new StringBuilder();

            if(i % 3 == 0) sb.append("Fizz");
            if(i % 5 == 0) sb.append("Buzz");
            if(sb.length() == 0) sb.append("" + i);

            list.add(sb.toString());
        }

        return list;
    }
}

File: cpp-algorithms/fizzbuzz.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cassert>
using namespace std;

class FizzBuzz {
public:
    vector<string> fizzbuzz(int n){
        vector<string> fizzBuzz;

        for(int i = 1; i <= n; i++) {
            if(i % 3 == 0 && i % 5 == 0)
                fizzBuzz.push_back("FizzBuzz");
            else if(i % 3 == 0)
                fizzBuzz.push_back("Fizz");
            else if(i % 5 == 0)
                fizzBuzz.push_back("Buzz");
            else
                fizzBuzz.push_back(to_string(i));
        }

        return fizzBuzz;
    }
};

int main() {
    FizzBuzz fb;
    int n = 15;
    vector<string> expected = {
        "1", "2", "Fizz", "4", "Buzz", "Fizz", "7", "8", "Fizz", "Buzz",
        "11", "Fizz", "13", "14", "FizzBuzz"
    };
    vector<string> result = fb.fizzbuzz(n);
    assert(result == expected);
    cout << "Test passed!" << endl;

    return 0;
}
